org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandler.java

```diff
@@ -233,7 +231,7 @@ private void addConstructorProposals(SignatureHelp help, ASTNode node, IMethod c
 						if (method.isConstructor()) {
 							InternalCompletionProposal proposal = new ConstructorProposal(CompletionProposal.METHOD_REF, pos);
 							proposal.setName(method.getElementName().toCharArray());
-							String signature = method.getSignature();
+							String signature = method.getSignature().replace('/', '.');

```

Comment:
Do you have an example of what this fixes ?

Reply:
See https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandlerTest.java#L376

Reply:
Ok. So the problem happens in `JavaModelUtil.isSameMethodSignature(..)` . It can't compare `java/lang/String` to `String`.